# Variant calling in mixed samples

Viral, bacterial, organellar genomes are non-diploid. As a result resquencing studies cannot rely on established diploid variant calling practoces. Here we evaluate several simple varinat callers using a mixed sample generated by duplex-sequencing. Our goal was to identify variants in mixtures of multiple haplotypes sequenced at very high coverage. Such dataset are typical in modern bacterial and viral genomic studies. In addition, we are seeking to be able to detect variants with frequencies around the NGS detection threshold of ~ 1% ([Salk et al. 2018](http://dx.doi.org/10.1038/nrg.2017.117)). In order to achieve this goal we selected a test dataset, which is distinct from data used in recent method comparisons ([Bush et al. 2019](http://dx.doi.org/10.1101/653774); [Yoshimura et al. 2019](http://dx.doi.org/10.1099/mgen.0.000261)). These data originate from a duplex sequencing experiment recently performed by our group ([Mei et al. 2019](https://academic.oup.com/gbe/article/11/10/3022/5572121)).  In this dataset a population of *E. coli* cells transformed with pBR322 plasmid is maintained in a turbidostat culture for an extensive period of time. Adaptive changes accumulated within the plasmid are then revealed with duplex sequencing ([Schmitt et al. 2012](http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&dopt=Citation&list_uids=22853953)). Duplex sequencing allows identification of variants at very low frequencies. This is achieved by first tagging both ends of DNA fragments to be sequenced with unique barcodes and subjecting them to paired-end sequencing. After sequencing read pairs containing identical barcodes are assembled into families. This procedure allows to reliably separate errors introduced during library preparation and/or sequencing (present in some but not all members of a read family) from true variants (present in all members of a read family derived from both strands).

For the following analysis we selected two data points from [Mei et al. 2019](https://academic.oup.com/gbe/article/11/10/3022/5572121): one corresponding to the beginning of the experiment (s0) and the other to the end (s5). The first sample is expected to be nearly clonal with no variation, while the latter contains a number of adaptive changes with frequencies around 1%. We aligned duplex consensus sequences (DCS) against the pBR322. We then walked through read alignments to produce counts of non-reference bases at each position (Fig. 1). 

| Position | A+ | A- | C+ | C- |G+ | G- |T+ | T- |N+ | N- | Del | Ins | 
|---------|----:|-----:|----:|----:|----:|----:|----:|----:|---:|---:|---:|---:|
| 3029 | 5 | 7 | 0 | 0 | 4 | 1 | 5018 | 6417 | 4 | 2 |  0 | 0 | 
| 3030 | 2 | 3 | 2 | 4 | 5010 | 6432 | 0 | 0 | 0 | 13 | 0 | 0 | 
| 3031 | 2 | 6 | 3 | 0 | 5005 | 6438 | 0 | 0 | 0 | 3 |  0 | 0 | 
| 3032 | 0 | 1 | 10 | 21 | 1 |  1 | 5011 | 6386 | 2 | 8 | 3 | 0 |
| 3033 | 4948 | 6367 | 32 | 39 | 2 | 3 | 2 | 8 | 1 | 3 | 0 | 0 | 
| 3034 | 0 | 0 | 2 | 1 | 4987 | 6374 | 1 | 1 | 0 | 7 | 0 | 0 |
| 3035 | 1 | 0 | 5015 | 6370 | 0 | 0 | 2 | 1 | 1 | 9 | 0 | 0 | 
| 3118 | 5 | 11 | 5121 | 6032 | 0 | 0 | 16 | 24 | 1 | 2 | 0 | 0 |

![image](https://github.com/user-attachments/assets/86dda763-5796-48a0-befe-6497bb989709)




